resolwe_bio/filters.py

```diff
@@ -32,6 +34,56 @@ class BioCollectionFilter(CollectionFilter):
     sample = filters.ModelChoiceFilter(
         field_name="entity", queryset=Sample.objects.all()
     )
+    descriptor__general__species = filters.CharFilter(
+        field_name="descriptor__general__species", lookup_expr="icontains"
+    )
+    tissue_type = filters.CharFilter(method="_filter_by_tissue")
+    treatment = filters.CharFilter(method="_filter_by_treatment")
+    outcome_is_defined = filters.CharFilter(method="_filter_by_outcome")
+
+    def _filter_by_multiple_fields(self, queryset, fields, value):
+        """Filter by any of the given fields.
+
+        Return exactly those objects, who icontain `value` in at least one of
+        `fields`.
+        """
+
+        query = Q()
+        for field in fields:
+            query |= Q(**{field + "__icontains": value})
+
+        return queryset.filter(query)
+
+    def _filter_by_tissue(self, queryset, name, value):
+        locations = [
+            "descriptor__general__organ",
+            "descriptor__general__biosample_source",
+            "descriptor__disease_information__organ_part",
+            "descriptor__disease_information__biopsy_site",
+            "descriptor__pathological_information__organ_part",
+            "descriptor__pathological_information__biopsy_site",
+        ]
+        return self._filter_by_descriptor_fields(queryset, locations, value)
+
+    def _filter_by_treatment(self, queryset, name, value):
+        locations = [
+            "descriptor__general__biosample_treatment",
+            "descriptor__treatment_type__drug",
+            "descriptor__immuno_oncology_treatment_type__io_drug",
+        ]
+        return self._filter_by_descriptor_fields(queryset, locations, value)
+
+    def _filter_by_outcome(self, queryset, name, value):
+        locations = [
+            "descriptor__response_and_survival_analysis__clinical_benefit",
+            "descriptor__response_and_survival_analysis__confirmed_bor",
+            "descriptor__response_and_survival_analysis__unconfirmed_bor",
+            "descriptor__response_and_survival_analysis__pfs",
+            "descriptor__response_and_survival_analysis__os",
+            "descriptor__response_and_survival_analysis__dfs",
+            "descriptor__response_and_survival_analysis__ttp",
```

Comment:
I believe you should also add corresponding indices in the database, otherwise, these queries will be slow.

See https://github.com/genialis/resolwe-bio/blob/master/resolwe_bio/migrations/resolwe_indexes_2.sql for example how to add missing indices to JSON fields. 

Check also that the added indices are really applied to the query.